examples/apps/cli/main.c

```diff
@@ -50,12 +50,12 @@
 extern void otAppCliInit(otInstance *aInstance);
 
 #if OPENTHREAD_CONFIG_HEAP_EXTERNAL_ENABLE
-void *otPlatCAlloc(size_t aNum, size_t aSize)
+OT_TOOL_WEAK void *otPlatCAlloc(size_t aNum, size_t aSize)
```

Comment:
Not probably to do this in this PR, but more general question is if it makes sense to remove these plat implementation from `main.c` (move them to the underlying platform).

Reply:
I had also thought to remove them because they are supposed to be platform specific functions.

Changing these implementations to weakly defined functions alleviates the problem, but only for platforms that have defined their platform library target as an `OBJECT` library.

Building the example apps targeting platforms with that use a `STATIC` library target may end up taking the these default implementations instead of a platform specific implementation.

For example, currently on the ot-efr32 main branch, if you look at the .map file for ot-cli-ftd, you'll see that the linker ended up using these functions instead of the ones defined in https://github.com/openthread/ot-efr32/blob/main/src/src/memory.c

I could change this PR to just remove the functions instead of making them weak. 

Thoughts?

Reply:
Thanks @lmnotran. Either way is fine for me (keeping this PR as is and maybe taking care of it later in separate PR, or take care of it from this PR).

Reply:
Let's keep it as is in this PR. Feel free to merge whenever you're ready